library/src/test/java/com/novoda/downloadmanager/lib/CriteriaTest.java

```diff
@@ -0,0 +1,180 @@
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_AND_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenTwoSelectionsWhenWeBuildWithOrThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .and()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_COLUMN_OR_ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY)
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithInnerCriteriaWhenWeBuildThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                .withArgument(ARGUMENT)
+                .or()
+                .withInnerCriteria(
+                        new Criteria.Builder()
+                                .withSelection(ANOTHER_SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .and()
+                                .withSelection(SELECTION_COLUMN, Criteria.Wildcard.EQUALS)
+                                .withArgument(ARGUMENT)
+                                .build())
+                .build();
+
+        CriteriaAssert.assertThat(criteria)
+                .hasSelection(SELECTION_QUERY + OR + "(" + ANOTHER_SELECTION_COLUMN_QUERY + AND + SELECTION_QUERY + ")")
+                .hasArguments(new String[]{ARGUMENT, ARGUMENT, ARGUMENT});
+    }
+
+    @Test
+    public void givenMultipleSelectionsWithMultipleInnerCriteriaWhenBuildingThenTheSelectionQueryIsCorrect() {
+        Criteria criteria = new Criteria.Builder()
+                .withInnerCriteria(
```

Comment:
@amlcurran as @friedger mentions the `innerCritera` is just an inner selection as in: 
`SELECT * from TABLE WHERE column1=1 OR (colum2=2 AND column3=3)`

here `(colum2=2 AND column3=3)` is an _inner criteria_ as it is contained within another criteria. This enables us to favour the priority of the criteria evaluation as the inner one will be evaluated first. 

I don't know what to say about `inBrackets()`, I wanted to emphasise the composable nature of the criterias (being able to construct one within another).